cmd/root.go

```diff
@@ -51,6 +60,78 @@ func init() {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 }
 
+// isArgTuneable returns if the argument provided is allowed to be modified
+func isArgTunable(arg string) bool {
+	return tuneableArgsWhitelist[arg]
+}
+
+// parseTuningFile parses the kernel argument tuning file
+func parseTuningFile(tuningFilePath string) (addArguments []types.TuneArgument, err error) {
+	if tuningFilePath == "" {
+		tuningFilePath = kernelTuningFile
+	}
+	// Return fast if the file does not exist
+	if _, err := os.Stat(tuningFilePath); os.IsNotExist(err) {
+		glog.V(2).Infof("no kernel tuning needed as %s does not exist", tuningFilePath)
+		// This isn't an error. Return out.
+		return addArguments, err
+	}
+	// Read and parse the file
+	file, err := os.Open(tuningFilePath)
+	// Clean up
+	defer func() {
+		closeErr := file.Close()
+		if closeErr != nil {
+			err = closeErr
+		}
+	}()
+	if err != nil {
+		// If we have an issue reading return an error
+		glog.Infof("Unable to open %s for reading: %v", tuningFilePath, err)
+		return addArguments, err
+	}
+
+	// Parse the tuning lines
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "ADD") {
```

Comment:
For this type of stuff, I usually include the trailing whitespace, e.g. `"ADD "` to be explicit and not mistakenly match e.g. `ADDFOO`. (And then `line[len("ADD ")]` under).